design-documents/service-isolation/cicd-pipeline.md

```diff
@@ -0,0 +1,48 @@
+# CI/CD pipeline for service isolation
+
+## About
+
+In the scope of service isolation we want to have CI/CD process that would allow
+* Running tests on change set and update preview environment
+* Automate PR delivery to mainline
+
+And in addition to that support the following scenarios that can be used internally and by system integrators
+* Update stage environment
+* Update production environment
+* Support blue/green deployment
+
+Steps that are currently not automated
+* Creating builds for change set
+* Updating preview environment (teams use for testing)
```

Comment:
If using [Netlify](https://www.netlify.com/), updating the PR with a new commit or doing a force push would trigger an automated preview environment build.